### DISCUSSION TITLE
How to skip the model_has_permissions query when no directly assigned permissions are present?

### DISCUSSION THREAD

**Discussion by anwarx4u** [spatie/laravel-permission#2351] [Q&A] [answered]
Hello,

I am using the awesome package in some of my applications, and it's working perfectly. However, I noticed that none of these applications are using the direct permission assignment feature, but the package is still querying the model_has_permissions table. I was wondering if there is a setting or hook that could allow us to skip this query altogether.

Thank you in advance.

![image](https://user-images.githubusercontent.com/21153921/223932553-8d1b9137-beed-468a-b084-eaf5b9ae6281.png)

**Comment by anwarx4u** [comment, marked as answer]
I have successfully overridden the HasPermissions trait in the user model using the following code, and it is currently functioning as intended. However, I would like to know if there is a better way to achieve this. 

```php
public function getAllPermissions(): Collection
    {
  
       $permissions = $this->permissions; // replaced by below line
        $permissions = collect();
       if (method_exists($this, 'roles')) {
            $permissions = $permissions->merge($this->getPermissionsViaRoles());
        }

        return $permissions->sort()->values();
    }

    public function hasDirectPermission($permission): bool
    {
        return false;
    }
```

**Reply by drbyte** [reply]
Yes, that should be fine.

**Reply by emmanuelsoto22** [reply]
Ok but I have differents question, It's mandatory use model_has_roles table in laravel permission?, It's posible skip it?

**Reply by drbyte** [reply]
Please open a new discussion, and explain why you want to skip it.
